src/wormhole/session/item/rename_gotchi.rs

```diff
@@ -0,0 +1,95 @@
+use super::SessSend;
+use hcor::id::ItemId;
+use hcor::{id, item, ConfigError, Item};
+use std::fmt;
+
+#[derive(Debug)]
+pub enum Error {
+    NoSuchItemConf(ConfigError),
+    NoSuch(id::NoSuch),
+}
+use Error::*;
+
+impl From<id::NoSuch> for Error {
+    fn from(ns: id::NoSuch) -> Error {
+        Error::NoSuch(ns)
+    }
+}
+
+impl From<hcor::ConfigError> for Error {
+    fn from(e: hcor::ConfigError) -> Error {
+        Error::NoSuchItemConf(e)
+    }
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "couldn't spawn items: ")?;
+        match self {
+            NoSuchItemConf(e) => write!(f, "no such item conf: {}", e),
+            NoSuch(e) => write!(f, "no such item: {}", e),
+        }
+    }
+}
+
+pub fn rename(ss: &mut SessSend, item_id: ItemId, new_name: String) -> Result<String, Error> {
+    let item = ss.item_mut(item_id)?;
+    let gotchi = item.gotchi_mut()?;
+    gotchi.nickname = new_name.clone();
+    Ok(new_name.clone())
+}
+
+#[cfg(all(test, features = "hcor_client"))]
+mod test {
+    #[actix_rt::test]
+    /// NOTE: requires that at least one item exists in the config!
+    async fn spawn() -> hcor::ClientResult<()> {
+        use super::test::{ITEM_ARCHETYPE, ITEM_SPAWN_COUNT};
+        use hcor::Hackstead;
+        use log::*;
+
+        // attempt to establish logging, do nothing if it fails
+        // (it probably fails because it's already been established in another test)
+        drop(pretty_env_logger::try_init());
+
+        debug!("create bob's stead");
+        let mut bobstead = Hackstead::register().await?;
+
+        // we'll need to keep track of how many items we have to see if spawning works.
+        fn count_relevant_items(hackstead: &Hackstead) -> usize {
```

Comment:
yes, this test definitely tests renaming gotchi and not, like, spawning items. yes.